packages/app-content-pages/src/shared/components/ZooHeaderWrapper/ZooHeaderWrapperContainer.js

```diff
@@ -4,72 +4,94 @@ import { withRouter } from 'next/router'
 import auth from 'panoptes-client/lib/auth'
 import { bool, func, shape, string } from 'prop-types'
 import { Component } from 'react'
-import Url from 'url-parse'
 
-function storeMapper (stores) {
+function storeMapper(stores) {
   const { user } = stores.store
   return {
     user
   }
 }
 
 class ZooHeaderWrapperContainer extends Component {
-  constructor () {
+  constructor() {
     super()
     this.openRegisterModal = this.openRegisterModal.bind(this)
     this.openSignInModal = this.openSignInModal.bind(this)
     this.signOut = this.signOut.bind(this)
   }
 
-  createUserProp () {
+  createUserProp() {
     const { user } = this.props
     return (user.isLoggedIn)
-      ? { 'display_name': user['display_name'] }
+      ? { display_name: user.display_name, login: user.login }
       : {}
   }
 
-  getUrlObject () {
-    const { asPath } = this.props.router
-    return new Url(asPath, true)
+  getUrlObject() {
+    return new URL(window.location)
   }
 
-  addUrlQuery (urlObject, propertyToAdd) {
-    const query = Object.assign({}, urlObject.query, {
-      [propertyToAdd]: 'true'
-    })
-    urlObject.set('query', query)
+  addUrlQuery(urlObject, propertyToAdd) {
+    urlObject.searchParams.set(propertyToAdd, true)
   }
 
-  redirect (urlObject) {
-    const { pathname, push } = this.props.router
-    const urlString = urlObject.toString().substr(urlObject.origin.length)
-    push(pathname, urlString, { shallow: true })
+  redirect(urlObject) {
+    const { push } = this.props.router
+    const urlString = urlObject.toString()
+    push(urlString, urlString, { shallow: true })
```

Comment:
I’m not 100% sure, but I think you can pass URL objects to `router.push` without converting them to strings. That would simplify the code.